### PR TITLE
SI-4015 Mixin and specialization work well

### PR DESCRIPTION
**Pull request by VladUreche**
The bug was fixed along with SI-7638 in 504b5f3.

Review by @axel22.

### CODE REVIEW

**Comment by VladUreche** [comment]
SI-6265 still pending, `abstract override` leads to duplicate methods with the same signature:
https://issues.scala-lang.org/browse/SI-6265?focusedCommentId=65706&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-65706

File: test/files/pos/SI-4012-b.scala [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+object Sub extends Super[Int] {
+  // it is expected that super[Super].superb crashes, since 
+  // specialization does parent class rewiring, and the super
+  // of Sub becomes Super$mcII$sp and not Super. But I consider
+  // this normal behavior -- if you want, I can modify duplicatiors
+  // to make this work, but I consider it's best to keep this
+  // let the user know Super is not the superclass anymore.
+  // super[Super].superb - Vlad
```
Comment (VladUreche) [inline test/files/pos/SI-4012-b.scala:12]: @axel22, do you think this should get rewired in duplicators?

**Comment by gkossakowski** [comment]
@VladUreche: Thanks a lot for the test! Could you grab @axel22's attention to review it?

**Comment by gkossakowski** [comment]
@VladUreche, @axel22: ping

**Comment by retronym** [comment]
Closing for now. The submitter bears the responsibility to seek review; @VladUreche when you return from vacation please reopen.

**Comment by VladUreche** [comment]
@axel22: when this gets to the top of your stack, please review so @gkossakowski can merge it.

**Pull request closed without merging**